extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -196,7 +196,10 @@ void configureRuntimeProperties(AgroalRecorder recorder,
 
         recorder.configureRuntimeProperties(agroalRuntimeConfig);
 
-        dataSourceInitialized.produce(new DataSourceInitializedBuildItem());
+        Set<String> dataSourceNames = agroalBuildTimeConfig.namedDataSources.keySet();
+        DataSourceInitializedBuildItem buildItem = new DataSourceInitializedBuildItem(dataSourceNames);
```

Comment:
I would prefer having `DataSourceInitializedBuildItem` totally immutable.

Immutable objects are nice :).

Reply:
Yeah, right, immutable objects are nice. It may be a bit irritating, but it is immutable.
"withDefaultDatasourceIfPresent" and "withDefaultDatasourceIfPresent" return new objects, if state changes. 

Let me know, if i should express it clearer e.g. by introducing a Builder.

Reply:
Why don't you simply have 2 different constructors? That would be simpler.